API/src/main/java/de/brandwatch/minianalytics/api/security/service/CustomUserDetailsService.java

```diff
@@ -0,0 +1,39 @@
+package de.brandwatch.minianalytics.api.security.service;
+
+import de.brandwatch.minianalytics.api.security.model.Role;
+import de.brandwatch.minianalytics.api.security.model.User;
+import de.brandwatch.minianalytics.api.security.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    @Autowired
+    public CustomUserDetailsService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userRepository.findByUsername(username);
+        if(user == null) throw new UsernameNotFoundException(username);
```

Comment:
formatting issue. put throw in another line

Reply:
Changed.